cmd/ethrex_replay/src/cli.rs

```diff
@@ -624,7 +650,12 @@ async fn replay_block(block_opts: BlockOptions) -> eyre::Result<()> {
 
     let (eth_client, network) = setup(&opts).await?;
 
-    let cache = get_blockdata(eth_client, network.clone(), or_latest(block)?).await?;
+    #[cfg(not(feature = "l2"))]
+    let fee_vault = None;
+    #[cfg(feature = "l2")]
+    let fee_vault = get_fee_vault_address(&eth_client).await?;
```

Comment:
We should probably change this to be a runtime flag, but using the `cfg!` macro instead of the attribute is a good first step. That way, the code is still checked, and unused code is removed automatically by the compiler, reducing "unused code" warnings.

```suggestion
    if cfg!(feature = "l2") {
        let fee_vault = get_fee_vault_address(&eth_client).await?;
    } else {
        let fee_vault = None;
    }
```